doc/holochain_101/src/zome/zome_functions.md

```diff
@@ -99,9 +73,22 @@ The `name` is the most important thing here, because when a function call to an
 
 ## Data Interchange - Inputs and Outputs
 
-In order to maintain compabitility with a variety of languages, it was decided to use a language agnostic data interchange format for inputs and ouputs. JSON, the modern web format was selected. Other formats may be supported in the future.
+In order to support building zomes in a variety of languages, we decided to use a simple language agnostic function specification format, using JSON.  Other formats may be supported in the future.
+
+This has two big implications: Holochain Conductor implementations must handle JSON serialization and deserialization on the "outside", and HDKs and Zomes must handle JSON serialization and deserialization on the "inside". Holochain agrees only to mediate between the two by passing a string (which should represent valid JSON data).
 
-Put simply, this has two big implications: Holochain Conductor implementations must handle JSON serialization and deserialization on the "outside", and HDKs and Zomes must handle JSON serialization and deserialization on the "inside". Holochain agrees only to mediate between the two by passing a string (which should represent valid JSON data).
+## Traits
+Traits provide a way to group functions by name.  The primary use of this feature is for creating a composibility space where DNA creators can implement different DNAs to emergent function interfaces and then compose with them in the conductor by matching on the function group names and signatures.  Additionally Holochain may reserve a few special trait names that have specific side-effects.  The first of such reserved names is `hc_public`.  Functions grouped in this name will have automatically added to a public capability grant that happens at genesis time, thus making them accessible to any caller.  For more details on the Holochain security model please see the [Capbilities](capabilities.md) section.
+
+Here is an example of what a trait definition using the public reserved trait name might look like:
+
+```json
+"traits": {
+    "hc_public": {
+        "functions": ["get_task_list"]
+    }
+}
```

Comment:
This doesn't match the code. Shouldn't it be:
```
traits: {
  hc_public [get_task_list]
}
```
?

Reply:
That file shows the json that comes out at the end.  Not the rust code.